src/extension.ts

```diff
@@ -34,42 +39,37 @@ class VerySmartSelect {
         this.strategies["json"] = new TypescriptStrategy();
 
         this.windowSelectionListener = window.onDidChangeTextEditorSelection(e => {
-            const newSelection = e.selections[0];
-            if (this.lastRange === undefined
-                || this.lastRange.start !== e.textEditor.document.offsetAt(newSelection.start)
-                || this.lastRange.end !== e.textEditor.document.offsetAt(newSelection.end)) {
-                this.lastRange = undefined;
-                this.selectionHistory = [];
+            if (this.didUpdateSelections) {
+                this.didUpdateSelections = false;
```

Comment:
Overlapping selections appeared to get coalesced, so opt-ed for this simpler check